### DISCUSSION TITLE
Best practices for filtering and QC with scATAC-seq data

### DISCUSSION THREAD

**Discussion by smk5g5** [stuart-lab/signac#418] [Q&A] [answered]
Hi,

I have tried to google if I can find how people usually do filtering and QC for scATAC-seq data. I don't have much experience with scATAC-seq data but have worked with scRNA-seq data for which there are a plethora of online posts available on how to QC and filter the data`http://bioconductor.org/books/release/OSCA/` "Orchestrating Single-Cell Analysis with Bioconductor" being one of these. So I was wondering if you can give examples on what you think are the best practices for filtering and QC with scATAC-seq data (for multiple samples) starting with the cellranger output. I did find this lecture of Dr. Liu informative (https://www.youtube.com/watch?v=r39Ux0vKDgU&list=PLeB-Dlq-v6tY3QLdQBA7rwb4a7fK9mLpv&index=54) in this regard but they use other tools like Maestro etc which have also been used for Bulk-ATACseq to accomplish this. I was wondering if you guys have the same suggestions. Any inputs would be much appreciated. Thanks!

**Comment by timoast** [comment, marked as answer]
We recommend filtering cells based on some of the QC metrics shown in the Signac [vignettes](https://satijalab.org/signac/articles/overview.html). These include:

* Minimum and maximum thresholds for total counts per cell
* TSS enrichment score
* Nucleosome banding pattern score
* Fraction of counts in genomic blacklist regions

To set a cutoff for each of these metrics, it's usually best to look at the distribution of values in your dataset and choose a reasonable value that would remove outliers for each metric. It's also important to examine each of these metrics with cells projected into a low-dimensional space (eg, UMAP), and check if you see cells grouping together that are outliers for any of the QC metrics (for example, do you find a group of high blacklist fraction cells?), and if so you might need to adjust the cutoffs used. 

Another QC step to consider is doublet detection. We currently don't have functions for this in Signac, but you can try to identify these post-hoc in your data by identifying clusters that contain peaks that are not normally accessible in the same cell type. You might also be able to adapt some of the computational doublet detection methods originally designed for scRNA-seq to work with scATAC-seq data.

**Reply by smk5g5** [reply]
Thanks for this input. I was wondering in the "merging objects" documentation why are you filtering out peaks based on this criteria: 

```# Filter out bad peaks based on length
peakwidths <- width(combined.peaks)
combined.peaks <- combined.peaks[peakwidths  < 10000 & peakwidths > 20]
combined.peaks
```
Also, the correct approach for filtering multiple samples of a scATAC-seq batch would look like this https://github.com/atimms/ratchet_scripts/blob/b1a722c0d65a64075de8363fba82b31861bec05c/cherry_sc_org_manuscript/cherry_human_scATAC_all_0620.R or is there a better way? Thanks!

**Reply by smk5g5** [reply]
Also how did you decide  on 
```min.cells = 10,
min.features = 200``` 
 for the `CreateChromatinAssay` function in the vignette?

**Reply by timoast** [reply]
In the vignette we remove those peaks that are outliers in their width, as these may represent inaccurate peak calls (implausibly short or long for ATAC-seq peaks).

The minimum number of features to used to retain a cells (`min.features`) is similar to the metrics I mentioned above (minimum and maximum thresholds for total counts per cell).
